extensions/csrf-reactive/runtime/src/main/java/io/quarkus/csrf/reactive/runtime/CsrfRequestResponseReactiveFilter.java

```diff
@@ -96,18 +96,21 @@ public void filter(ResteasyReactiveContainerRequestContext requestContext, Routi
             }
         } else if (config.verifyToken) {
             // unsafe HTTP method, token is required
-
-            if (!isMatchingMediaType(requestContext.getMediaType(), MediaType.APPLICATION_FORM_URLENCODED_TYPE)
-                    && !isMatchingMediaType(requestContext.getMediaType(), MediaType.MULTIPART_FORM_DATA_TYPE)) {
+            MediaType mediaType = requestContext.getMediaType();
+            // If no media type is sent, assume application/x-www-form-urlencoded
+            if (mediaType == null) {
+                mediaType = MediaType.APPLICATION_FORM_URLENCODED_TYPE;
+            }
```

Comment:
Are you sure about that?

Reply:
About what exactly?

Reply:
Setting the default media type to this value?

Reply:
Sorry maybe it's not clear enough: if the media type is not set, I would expect it to not be compatible. But maybe you want to try the checks anyway?

TBH, I'm not sure what's the best outcome here and given it is related to security, maybe we should ask @sberyozkin to have a look.

Reply:
Ah I thought you were talking about the media type being null, I've removed that and tested against a null value